src/Conversions/Commands/RegenerateCommand.php

```diff
@@ -75,17 +77,22 @@ public function handle(MediaRepository $mediaRepository, FileManipulator $fileMa
     public function getMediaToBeRegenerated(): Collection
     {
         $modelType = $this->argument('modelType') ?? '';
-        $mediaIds = $this->getMediaIds();
+        if ($modelType !== '') {
+            return $this->mediaRepository->getByModelType($modelType);
+        }
 
-        if ($modelType === '' && count($mediaIds) === 0) {
-            return $this->mediaRepository->all();
+        $startingFromId = (int)$this->option('starting-from-id');
+        if ($startingFromId !== 0) {
+            $excludeStartingId = $this->option('exclude-starting-id') ?? false;
+            return $this->mediaRepository->getByIdGreaterThan($startingFromId, $excludeStartingId);
         }
 
-        if (! count($mediaIds)) {
-            return $this->mediaRepository->getByModelType($modelType);
+        $mediaIds = $this->getMediaIds();
+        if (count($mediaIds) > 0) {
+            return $this->mediaRepository->getByIds($mediaIds);
         }
 
-        return $this->mediaRepository->getByIds($mediaIds);
+        return $this->mediaRepository->all();
```

Comment:
I think this is what you were suggesting? Handle the special cases first, then fall back to ids or all media

Reply:
This seems better now.

A case that isn't handled well now is we you pass both `modelType` and `startingFromId`. Right now that `startingFromId` will be ignored.

Reply:
This should work without refactoring the existing methods to pass the builder instead of the result and chaining everything depending on the args and options. 

I didn't think passing `ids` and `modelType` at the same time would make a ton of sense as a use case as the ids may not belong to the passed type.